catalog/ui/catalog-ui-search/src/main/webapp/component/query-schedule/query-schedule.view.js

```diff
@@ -51,51 +52,51 @@ define([
             }
         },
         setupInterval: function(){
-            var halfHour = 30 * 60 * 1000;
-            var hour = 2 * halfHour;
             this.propertyInterval.show(new PropertyView({
-                model: new Property({
-                    enum: [
-                        {
-                            label: 'Never',
-                            value: false
-                        },
-                        {
-                            label: '1/2 Hour',
-                            value: halfHour
-                        },
-                        {
-                            label: '1 Hour',
-                            value: hour
-                        },
-                        {
-                            label: '2 Hours',
-                            value: 2 * hour
-                        },
-                        {
-                            label: '4 Hours',
-                            value: 4 * hour
-                        },
-                        {
-                            label: '8 Hours',
-                            value: 8 * hour
-                        },
-                        {
-                            label: '16 Hours',
-                            value: 16 * hour
-                        },
-                        {
-                            label: 'Day',
-                            value: 24 * hour
-                        }
-                    ],
-                    value: [this.model.get('polling') || false],
-                    id: 'Frequency'
-                })
+                model: this.getPropertyIntervalEnum()
             }));
             this.propertyInterval.currentView.turnOffEditing();
             this.propertyInterval.currentView.turnOnLimitedWidth();
         },
+        getPropertyIntervalEnum: function() {
+            var intervalArray;
+            intervalArray = [{
+                label: 'Never',
+                value: false
+            }];
+
+            var that = this;
+            _.each(properties.scheduleFrequencyList, function(property) {
```

Comment:
Can we add ordering here since the values can be added arbitrarily to the metatype?